src/eval.c

```diff
@@ -488,7 +488,8 @@ void resetLuaClient(void) {
 
 void evalGenericCommand(client *c, int evalsha) {
     lua_State *lua = lctx.lua;
-    char funcname[43];
+    char funcname_buf[43];
+    char *funcname = funcname_buf;
```

Comment:
No longer cache `funcname`, forget to revert this?

Reply:
@sundb this is new, i needed that since i sometimes need the buffer to store into, and other times i use pre-computed name, so i needed a variable that can be changed to point to different buffers.
maybe i'm missing your point?

Reply:
But I can't see it being pointed to other buffers.
In commit [`f5344cf` (#11521)](https://github.com/redis/redis/pull/11521/commits/f5344cf4f5871a993d7040f941d9c6d36fdf7140) `funcname` points to c->eval_funcname, but in the later commit you remove it.

Reply:
it is by default (at init) pointing to `funcname_buf`, and in case `c->cur_script` is NULL, we pass it to `evalCalcFunctionName`

Reply:
Not sure if I missed something.

> @sundb this is new, i needed that since i sometimes need the buffer to store into, and other times i use pre-computed name, so i needed a variable that can be changed to point to different buffers. maybe i'm missing your point?

Does the `pre-computed name` refer to the [`f5344cf` (#11521)](https://github.com/redis/redis/pull/11521/commits/f5344cf4f5871a993d7040f941d9c6d36fdf7140) commit? But I see that you have reverted them in the [`27665e8` (#11521)](https://github.com/redis/redis/pull/11521/commits/27665e814be7969309de028a44ef3d6965f4d01b) commit.

If not, I think the following code should have the same effect.
```c
    char funcname [43];
    if (c->cur_script) {
        ...
    } else
        evalCalcFunctionName(evalsha, c->argv[1]->ptr, funcname);
```

Reply:
you're right. i must be blind

Reply:
```suggestion
    char funcname[43];
```